core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/configuration/security/jwt/JWTAuthenticationFilter.java

```diff
@@ -23,14 +25,26 @@ public JWTAuthenticationFilter(JWTAuthenticationConverter jwtAuthenticationConve
     }
 
     @Override
-    protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain) throws ServletException, IOException {
-        Authentication authentication = jwtAuthenticationConverter.convert(httpServletRequest);
-        if (authentication == null) {
```

Comment:
Is this line not needed (anymore). Do we need to check if it is possible that a null Authentication is returned by jwtAuthenticationConverter.convert?

Reply:
Wil give a 403 in any case

Reply:
not true. If no Authorization header is present in the request or the Authorization header does not start with Bearer, the authentication object will be null. This behavior is expected for endpoints that are whitelisted from authorization e.g. /actuator/health. That's why we just call the next filter in the chain. Also, should there have been other filters that have already set the authentication object of SecurityContextHolder.getContext(), there value will be overridden by a null value. This is not wanted behavior